pkg/mcs/scheduling/server/apis/v1/api.go

```diff
@@ -153,40 +173,100 @@ func getOperatorByID(c *gin.Context) {
 // @Failure  500  {string}  string  "PD server failed to proceed the request."
 // @Router   /operators [GET]
 func getOperators(c *gin.Context) {
-	svr := c.MustGet(multiservicesapi.ServiceContextKey).(*scheserver.Server)
+	handler := c.MustGet(handlerKey).(*handler.Handler)
 	var (
 		results []*operator.Operator
-		ops     []*operator.Operator
 		err     error
 	)
 
-	opController := svr.GetCoordinator().GetOperatorController()
-	if opController == nil {
-		c.String(http.StatusInternalServerError, err.Error())
-		return
-	}
 	kinds := c.QueryArray("kind")
 	if len(kinds) == 0 {
-		results = opController.GetOperators()
+		results, err = handler.GetOperators()
 	} else {
-		for _, kind := range kinds {
-			switch kind {
-			case "admin":
-				ops = opController.GetOperatorsOfKind(operator.OpAdmin)
-			case "leader":
-				ops = opController.GetOperatorsOfKind(operator.OpLeader)
-			case "region":
-				ops = opController.GetOperatorsOfKind(operator.OpRegion)
-			case "waiting":
-				ops = opController.GetWaitingOperators()
-			}
-			results = append(results, ops...)
-		}
+		results, err = handler.GetOperatorsByKinds(kinds)
 	}
 
+	if err != nil {
+		c.String(http.StatusInternalServerError, err.Error())
+		return
+	}
 	c.IndentedJSON(http.StatusOK, results)
 }
 
+// @Tags     operator
+// @Summary  Cancel a Region's pending operator.
+// @Param    region_id  path  int  true  "A Region's Id"
+// @Produce  json
+// @Success  200  {string}  string  "The pending operator is canceled."
+// @Failure  400  {string}  string  "The input is invalid."
+// @Failure  500  {string}  string  "PD server failed to proceed the request."
+// @Router   /operators/{region_id} [delete]
+func deleteOperatorByRegion(c *gin.Context) {
+	handler := c.MustGet(handlerKey).(*handler.Handler)
+	id := c.Param("id")
+
+	regionID, err := strconv.ParseUint(id, 10, 64)
+	if err != nil {
+		c.String(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	if err = handler.RemoveOperator(regionID); err != nil {
+		c.String(http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	c.String(http.StatusOK, "The pending operator is canceled.")
+}
+
+// @Tags     operator
+// @Summary  lists the finished operators since the given timestamp in second.
+// @Param    from  query  integer  false  "From Unix timestamp"
+// @Produce  json
+// @Success  200  {object}  []operator.OpRecord
+// @Failure  400  {string}  string  "The request is invalid."
+// @Failure  500  {string}  string  "PD server failed to proceed the request."
```

Comment:
Do we have it here?

Reply:
It is a copy from server/api, maybe we need to fix all other comments too.